src/chai/ArrayManager.cpp

```diff
@@ -215,50 +250,54 @@ void ArrayManager::allocate(
   pointer_record->m_pointers[space] =  alloc.allocate(size);
   registerPointer(pointer_record, space);
 
-  CHAI_LOG("ArrayManager", "Allocated array at: " << ret);
+  CHAI_LOG("ArrayManager", "Allocated array at: " << pointer_record->m_pointers[space]);
 }
 
-void ArrayManager::free(PointerRecord* pointer_record)
+void ArrayManager::free(PointerRecord* pointer_record, ExecutionSpace spaceToFree)
 {
   for (int space = CPU; space < NUM_EXECUTION_SPACES; ++space) {
-    if (pointer_record->m_pointers[space]) {
-      if (pointer_record->m_owned[space]) {
-        void* space_ptr = pointer_record->m_pointers[space];
+    if (space == spaceToFree || spaceToFree == NONE) {
+       if (pointer_record->m_pointers[space]) {
+         if (pointer_record->m_owned[space]) {
+           void* space_ptr = pointer_record->m_pointers[space];
 #if defined(CHAI_ENABLE_UM)
-        if (space_ptr == pointer_record->m_pointers[UM]) {
-          pointer_record->m_user_callback(ACTION_FREE,
-                                          ExecutionSpace(UM),
-                                          pointer_record->m_size);
-          m_pointer_map.erase(space_ptr);
-
-          auto alloc = m_resource_manager.getAllocator(
-              pointer_record->m_allocators[space]);
-          alloc.deallocate(space_ptr);
-
-          for (int space_t = CPU; space_t < NUM_EXECUTION_SPACES; ++space_t) {
-            if (space_ptr == pointer_record->m_pointers[space_t])
-              pointer_record->m_pointers[space_t] = nullptr;
-          }
-        } else {
+           if (space_ptr == pointer_record->m_pointers[UM]) {
```

Comment:
I hope these are not tabs!!!!

Reply:
No, just my editor is set to 3 spaces per tab instead of 2.

Reply:
3?!

Reply:
What!!! 3 spaces?!?